File: go/longest-digit-prefix/longest-digit-prefix.go
package main

import(
	"fmt"
)

func longestDigitsPrefix(inputString string)string{
	res := ""
	for _, val := range inputString{
		if val < 48 || val > 57{
			break
		}
		res += string(val)
	}
	return res
}

func main(){
	inputString := "123aa1"

	fmt.Printf("The longest digit prefix in the inputString = %s is: %s ", inputString, longestDigitsPrefix(inputString))

}

File: js/longest-digit-prefix.js
function longestDigitsPrefix(inputString){
    let res = ""
    for (const element of inputString) {
        if(!/\d/.test(element))
            break
        res += element
    }   
    return res
}

let inputString = "123aa1"

console.log(`The longest digit prefix in the inputString = ${inputString} is: `, longestDigitsPrefix(inputString))